rfcs/react-components/styles-handbook.md

```diff
@@ -0,0 +1,771 @@
+# Styles Handbook
+
+[@layeshifter](https://github.com/layershifter)
+
+This document covers how to efficiently use [Griffel][griffel] CSS-in-JS (used in Fluent UI React v9) to style components.
+
+## Table of contents
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Introduction](#introduction)
+- [Basics](#basics)
+- [APIs](#apis)
+  - [`makeStyles`](#makestyles)
+    - [Limitations](#limitations)
+  - [`mergeClasses()`](#mergeclasses)
+    - [Do not concatenate classes](#do-not-concatenate-classes)
+  - [`makeResetStyles`](#makeresetstyles)
+    - [Hybrid approach](#hybrid-approach)
+- [Advanced](#advanced)
+  - [Understanding selectors complexity](#understanding-selectors-complexity)
+- [Best practices](#best-practices)
+  - [Writting styles](#writting-styles)
+    - [Use `tokens` over direct colors](#use-tokens-over-direct-colors)
+    - [Avoid duplication in rule definitions](#avoid-duplication-in-rule-definitions)
+    - [Avoid `!important`](#avoid-important)
+  - [Performance](#performance)
+    - [Avoid nested `mergeClasses` calls](#avoid-nested-mergeclasses-calls)
+  - [Nested selectors](#nested-selectors)
+    - [Usage with pseudo classes](#usage-with-pseudo-classes)
+    - [Do not use selectors to target elements](#do-not-use-selectors-to-target-elements)
+    - [Do not introduce complicated selectors](#do-not-introduce-complicated-selectors)
+    - [Avoid usage of input pseudo classes](#avoid-usage-of-input-pseudo-classes)
+  - [RTL styles](#rtl-styles)
+    - [Using `@noflip`](#using-noflip)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# Introduction
+
+Griffel is a hybrid CSS-in-JS that features runtime option like any other CSS-in-JS and [Ahead-of-time compilation][griffel-aot] with [CSS extraction][griffel-css-extraction] to reduce runtime footprint and improve performance.
+
+# Basics
+
+Griffel uses Atomic CSS to generate classes. In Atomic CSS every property-value is written as a single CSS rule.
+
+```css
+/* Monolithic classes */
+/* Can be applied only to a specific button */
+.button {
+  display: flex;
+  align-items: center;
+}
+
+/* Atomic CSS */
+/* Can be applied to any element that needs there rules */
+.display-flex {
+  display: flex;
+}
+.align-items-center {
+  align-items: center;
+}
+```
+
+[Learn more][griffel-atomic-css] about Atomic CSS.
+
+# APIs
+
+## `makeStyles`
+
+Is used to define style permutations in components and is used for style overrides. Returns a [React hook][react-hook] that should be called inside a component:
+
+```jsx
+import { makeStyles } from '@griffel/react';
```

Comment:
doesn't fluent reexport these? since this guide is in fluent repo, I might expect the import to point to fluent.

Reply:
@jurokapsiar I added a note 3c10c6a.